Jeremy_Kracy/restAPI/api/routes/data.js

```diff
@@ -0,0 +1,108 @@
+const express = require('express');
+const router = express.Router();
+const mongoose = require('mongoose');
+
+const Data = require('../models/number');
+
+router.get('/', (req, res, next) => {
+    Data.find()
+    .select('data _id')
+    .exec()
+    .then(docs => {
+        res.status(200).json(docs);
+    })
+    .catch(err => {
+        console.log(err);
+        res.status(500).json({
+            error: err
+        })
+    });
+});
+
+router.get('/:dataId', (req, res, next) => {
+    const id = req.params.dataId;
+    Data.findById(id)
+    .select('data _id')
+    .exec()
+    .then(doc => {
+        if(doc) {
+            res.status(200).json(doc);
+        } else {
+            res.status(404).json({
+                message: "No valid entry for provided ID"
+            });
+        }
+    })
+    .catch(err => {
+        console.log(err);
+        res.status(500).json({
+            error: err
+        })
+    });
+});
+
+
+router.post('/', (req, res, next) => {
+    const data = new Data({
+        _id: new mongoose.Types.ObjectId(),
+        data: req.body.data
+    });
+    console.log(data.data.length);
+    if(data.data.length === 500){
+        data.save()
+        .then(result => {
+            console.log(result);
+            console.log(result.data.length);
+            
+            res.status(201).json({
```

Comment:
How did you decide on 201 here?

Reply:
I used the 201 status code to verify that a new object was created in the database using that POST request. This is in line with the definition of the "201 Created" code.